server/src/main/java/org/opensearch/indices/recovery/RecoverySourceHandler.java

```diff
@@ -720,7 +720,7 @@ void createRetentionLease(final long startingSeqNo, ActionListener<RetentionLeas
                 // it's possible that the primary has no retention lease yet if we are doing a rolling upgrade from a version before
                 // 7.4, and in that case we just create a lease using the local checkpoint of the safe commit which we're using for
                 // recovery as a conservative estimate for the global checkpoint.
-                assert shard.indexSettings().getIndexVersionCreated().before(LegacyESVersion.V_7_4_0)
+                assert shard.indexSettings().getIndexVersionCreated().before(LegacyESVersion.fromId(7040099))
```

Comment:
Do we actually need this assertion `shard.indexSettings().getIndexVersionCreated().before(LegacyESVersion.fromId(7040099))`? The 3.0.0 clustrer will never work with 7.x anyway, right?

Reply:
No... this can be removed. Thx!